File: src/upros_range/src/ros_sonic.cpp
#include <ros/ros.h>
#include <sensor_msgs/Range.h>

void rangeCallback1(const sensor_msgs::Range::ConstPtr& msg){
    ROS_INFO("Distance Front: %f", msg->range);
}

void rangeCallback2(const sensor_msgs::Range::ConstPtr& msg){
    ROS_INFO("Distance Left: %f", msg->range);
}

void rangeCallback3(const sensor_msgs::Range::ConstPtr& msg){
    ROS_INFO("Distance Right: %f", msg->range);
}

void rangeCallback4(const sensor_msgs::Range::ConstPtr& msg){
    ROS_INFO("Distance Back: %f", msg->range);
}

int main(int argc, char** argv){
    ros::init(argc, argv, "range_subscriber");
    ros::NodeHandle nh;
    ros::Subscriber sub_1 = nh.subscribe<sensor_msgs::Range>("/front_sonar", 10, rangeCallback1);
    ros::Subscriber sub_2 = nh.subscribe<sensor_msgs::Range>("/left_sonar", 10, rangeCallback2);
    ros::Subscriber sub_3 = nh.subscribe<sensor_msgs::Range>("/right_sonar", 10, rangeCallback3);
    ros::Subscriber sub_4 = nh.subscribe<sensor_msgs::Range>("/back_sonar", 10, rangeCallback4);
    ros::spin();
    return 0;
}

File: src/upros_range/scripts/ros_sonic.py
#!/usr/bin/env python3

import rospy
from sensor_msgs.msg import Range

def rangeCallback1(msg):
    rospy.loginfo("Distance Front: %f", msg.range)

def rangeCallback2(msg):
    rospy.loginfo("Distance Left: %f", msg.range)

def rangeCallback3(msg):
    rospy.loginfo("Distance Right: %f", msg.range)

def rangeCallback4(msg):
    rospy.loginfo("Distance Back: %f", msg.range)

if __name__ == '__main__':
    rospy.init_node('range_subscriber', anonymous=True)
    sub_1 = rospy.Subscriber('/front_sonar', Range, rangeCallback1)
    sub_2 = rospy.Subscriber('/left_sonar', Range, rangeCallback2)
    sub_3 = rospy.Subscriber('/right_sonar', Range, rangeCallback3)
    sub_4 = rospy.Subscriber('/back_sonar', Range, rangeCallback4)
    rospy.spin()